### PR TITLE
DDF-3660 Add Privileged Actions to URLResourceReader

### PR DESCRIPTION
**Pull request by peterhuffer**
#### What does this PR do?
- Add Privileged Actions to URLResourceReader
- Update CDM configurations.policy section
- Fixes a few SonarQube findings

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@clockard @emmberk @kcover 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here. If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl
@stustison

#### How should this be tested? (List steps with links to updated documentation)
Follow documentation to configure a CDM for in place directory and verify the ingested products can be downloaded via Intrigue.

#### Any background context you want to provide?
This change makes it so when downloading products ingested via the CDM  configured for a in place directory, that permissions do not need to be added for every type of endpoint that the product can be downloaded through. Originally, each endpoint would need to permissions to read from the configured directory being monitored from.

I'm not entirely sure if this is the correct place to do this privileged action or if this is safe.

#### What are the relevant tickets?
[DDF-3660](https://codice.atlassian.net/browse/DDF-3660)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java [hunk 1]
```diff
@@ -256,69 +261,72 @@ public ResourceResponse retrieveResource(URI resourceURI, Map<String, Serializab
...
-
-      final Serializable qualifierSerializable = properties.get(ContentItem.QUALIFIER_KEYWORD);
-      if (qualifierSerializable instanceof String) {
-        final String qualifier = (String) qualifierSerializable;
-        if (StringUtils.isNotBlank(qualifier)) {
-          resourceURI =
-              UriBuilder.fromUri(resourceURI)
-                  .queryParam(ContentItem.QUALIFIER_KEYWORD, qualifier)
-                  .build();
+    switch (resourceURI.getScheme()) {
```
Comment (peterhuffer) [inline catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java]: This section was just updated from if/else statements to a switch statement based on Intellij recommendation.

File: catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java [hunk 2]
```diff
@@ -256,69 +261,72 @@ public ResourceResponse retrieveResource(URI resourceURI, Map<String, Serializab
...
+      final String originalFileName = productName;
       productName =
-          StringUtils.defaultIfBlank(
-              handleContentDispositionHeader(
-                  connection.getHeaderField(HttpHeaders.CONTENT_DISPOSITION)),
-              productName);
+          AccessController.doPrivileged(
+              (PrivilegedAction<String>)
+                  () ->
+                      StringUtils.defaultIfBlank(
```
Comment (peterhuffer) [inline catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java]: I'm not entirely sure why this was the correct operation to do a privileged action on, and not when we get the input stream from the connection below in this same method.

File: catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java [hunk 3]
```diff
@@ -409,6 +410,16 @@ private ResourceResponse retrieveHttpProduct(
     }
   }
 
+  private String getResponseErrorMessage(InputStream is) {
```
Comment (peterhuffer) [inline catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java]: Made method to reduce complexity for SonarQube finding.

File: catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java (hunk 2 above)
Comment (stustison) [reply catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java]: I'm not sure why you'd need it around this block at all. The tika one makes sense, but not this one.

Comment (paouelle) [reply catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java]: ☝️

**Comment by clockard** [comment]
build now

File: catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java (hunk 2 above)
Comment (peterhuffer) [reply catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java]: There's a few lines below this where the input stream is read from the connection where I originally thought the problem would be and put the privileged action, but was still running into permission issues. I debugged through and found the issue here. Seems weird to me too.

**Comment by peterhuffer** [comment]
build now

**Comment by clockard** [comment]
Manual tested and verified. :shipit:

**Comment by peterhuffer** [comment]
yarn failure
build now

**Comment by clockard** [comment]
build now

**Pull request merged**